packages/arb-avm-cpp/avm/src/machinethread.cpp

```diff
@@ -48,6 +56,7 @@ bool MachineThread::runMachine(MachineExecutionConfig config,
 }
 
 bool MachineThread::continueRunningMachine(bool asynchronous) {
+    finishThread();
```

Comment:
Whenever we resolve the `runMachine` discussion, the same will apply here I think

Reply:
treating as fatal error